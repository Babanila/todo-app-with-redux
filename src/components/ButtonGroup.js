import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { cx, css } from 'emotion'
import SingleButton from './SingleButton'
import { showAllTodo, showCompletedTodo, showActiveTodo } from '../actions'

const btnDivStyles = css`
  margin-top: 1em;
  width: 60%;
  height: 2em;
  display: flex;
  justify-content: space-between;
`

function ButtonGroup() {
  const todos = useSelector((state) => state.todoReducer.todos)
  const dispatch = useDispatch()
  const handleShowAllTodo = () => dispatch(showAllTodo(todos))
  const handleShowCompletedTodo = () => dispatch(showCompletedTodo(todos))
  const handleShowActiveTodo = () => dispatch(showActiveTodo(todos))

  return (
    <div className={cx(btnDivStyles)}>
      <SingleButton btnName='All Todo' btnClick={handleShowAllTodo} />
      <SingleButton btnName='Completed Todo' btnClick={handleShowCompletedTodo} />
      <SingleButton btnName='Active Todo' btnClick={handleShowActiveTodo} />
    </div>
  )
}

export default ButtonGroup
