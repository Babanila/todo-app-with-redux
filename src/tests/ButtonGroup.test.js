import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import ButtonGroup from '../components/ButtonGroup'

const mockStore = configureStore([thunk])

describe('ButtonGroup', () => {
  let store, ButtonGroupWrapper

  beforeEach(() => {
    store = mockStore({
      todoReducer: {
        todos: []
      }
    })

    ButtonGroupWrapper = mount(
      <Provider store={store}>
        <ButtonGroup />
      </Provider>
    ).mount()
  })

  it('should be defined', () => {
    expect(ButtonGroup).toBeDefined()
  })

  it('should be a function', () => {
    expect(typeof ButtonGroup).toBe('function')
  })

  it('should renders site title correctly', () => {
    expect(ButtonGroupWrapper).toMatchSnapshot()
  })

  it('should render the exact number of element in component', () => {
    expect(ButtonGroupWrapper.find('div').length).toEqual(1)
    expect(ButtonGroupWrapper.find('SingleButton').length).toEqual(3)
  })
})
