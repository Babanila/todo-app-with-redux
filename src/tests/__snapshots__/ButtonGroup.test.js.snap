// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should renders site title correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ButtonGroup>
    <div
      className="css-12gomj1"
    >
      <SingleButton
        btnClick={[Function]}
        btnName="All Todo"
      >
        <button
          onClick={[Function]}
        >
          All Todo
        </button>
      </SingleButton>
      <SingleButton
        btnClick={[Function]}
        btnName="Completed Todo"
      >
        <button
          onClick={[Function]}
        >
          Completed Todo
        </button>
      </SingleButton>
      <SingleButton
        btnClick={[Function]}
        btnName="Active Todo"
      >
        <button
          onClick={[Function]}
        >
          Active Todo
        </button>
      </SingleButton>
    </div>
  </ButtonGroup>
</Provider>
`;
